ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DdlCommandExec.java

```diff
@@ -47,16 +48,21 @@ public DdlCommandExec(final MutableMetaStore metaStore) {
   /**
    * execute on metaStore
    */
-  public DdlCommandResult execute(final DdlCommand ddlCommand) {
-    return new Executor().execute(ddlCommand);
+  public DdlCommandResult execute(final String sql, final DdlCommand ddlCommand) {
+    return new Executor(sql).execute(ddlCommand);
   }
 
   private final class Executor implements io.confluent.ksql.execution.ddl.commands.Executor {
+    private final String sql;
+
+    private Executor(final String sql) {
+      this.sql = Objects.requireNonNull(sql, "sql");
+    }
 
     @Override
     public DdlCommandResult executeCreateStream(final CreateStreamCommand createStream) {
       final KsqlStream<?> ksqlStream = new KsqlStream<>(
-          createStream.getSqlExpression(),
+          sql,
```

Comment:
can we take this one step further and just remove it from `StructuredDataSource`? It doesn't seem to be used anywhere outside of tests, and even then just to compare equality.

Reply:
I hesitated to do that because it would be nice to return from the API (https://github.com/confluentinc/ksql/issues/2081).

Reply:
fair enough!